radixdlt/src/main/java/org/radix/Radix.java

```diff
@@ -162,23 +163,25 @@ private static void dumpExecutionLocation() {
 			String jarPath = jarFile;
 
 			if (jarPath.toLowerCase().endsWith(".jar")) {
-				jarPath = jarPath.substring(0, jarPath.lastIndexOf("/"));
+				jarPath = jarPath.substring(0, jarPath.lastIndexOf('/'));
 			}
 			System.setProperty("radix.jar.path", jarPath);
 
 			log.debug("Execution file: "+ System.getProperty("radix.jar"));
 			log.debug("Execution path: "+ System.getProperty("radix.jar.path"));
 		} catch (URISyntaxException | ClassNotFoundException e) {
-			throw new RuntimeException("while fetching execution location", e);
+			throw new IllegalStateException("Error while fetching execution location", e);
 		}
 	}
 
 	private static Universe extractUniverseFrom(RuntimeProperties properties, Serialization serialization) {
 		try {
 			byte[] bytes = Bytes.fromBase64String(properties.get("universe"));
-			return serialization.fromDson(bytes, Universe.class);
+			Universe u = serialization.fromDson(bytes, Universe.class);
+			UniverseValidator.validate(u);

```

Comment:
Unfortunately in the code reshuffle, it looks like the universe validation got lost, so it has been reinstated here.

Reply:
Whoops. Good catch.

Reply:
I know right?  :shame: